### PR TITLE
Fix trashManager not found

### PR DESCRIPTION
**Pull request by eneiluj**
Only use the trash manager if the files_trashbin app is enabled for the user.
Catch any exception when getting the trashManager, just in case.

https://community.openproject.org/wp/43936

### CODE REVIEW

**Comment by eneiluj** [comment]
The phpstan error can be ignored as it is the way to go :grin:.

File: lib/Controller/FilesController.php [hunk 1]
```diff
@@ -158,16 +160,24 @@ public function getFilesInfo(?array $fileIds): DataResponse {
...
 			$file = $files[0];
 			$trashed = false;
-		} else {
-			$file = $this->trashManager->getTrashNodeById(
-				$this->user, $fileId
-			);
-			$trashed = true;
+		} elseif ($this->appManager->isEnabledForUser('files_trashbin')) {
+			try {
+				$trashManager = Server::get(\OCA\Files_Trashbin\Trash\ITrashManager::class);
```
Comment (individual-it) [inline lib/Controller/FilesController.php:173]: this only works for NC 25, see https://github.com/nextcloud/server/commit/f945c0cbc6aef461fabf17cea42440ad9b6fea09#diff-e3c727c87edec3701539efc3b38c11374baf33362c0a4038f004f4b7ccbe71b7R39

That is why phpstan complains with older versions.

**Review by individual-it — CHANGES_REQUESTED** [review]

File: lib/Controller/FilesController.php (hunk 1 above)
Comment (eneiluj) [reply lib/Controller/FilesController.php]: Alright, fixed. Could you adjust the tests?

**Comment by individual-it** [comment]
strangely the unit tests work fine when run separately but when ran all at once `getTrashNodeById`  throw strange exceptions  `"Backends provided no user object for testUser"`
need to investigate more

**Comment by individual-it** [comment]
tests are fixed and some new ones added.
there are two cases
1. a file got deleted, then the trashbin app got disabled
    in that case the file actually is in the trashbin and has a mount, but cannot be accessed, the return code is 403
2. a file got deleted after the trashbin app got disabled
    in that case the file is really gone, the return code is 404

**Review by individual-it — COMMENTED** [review]
I think we can merge that
@eneiluj are you happy with my refactoring?
@wielinde should that be backported to 2.0 ?

**Comment by wielinde** [comment]
@individual-it Yes, I think this PR is a typical for a patch level release

**Comment by eneiluj** [comment]
@individual-it Yep, all good.

**Review by individual-it — APPROVED** [review]

**Comment by individual-it** [comment]
backport in  #232

**Pull request merged**